### ISSUE TITLE
Blazor WebAssembly Msal TryAddAdditionalParameter("prompt", "select_account") does nothing

### ISSUE THREAD

**Issue by juho-hanhimaki** [dotnet/aspnetcore#44854]
### Description

I am using the MSAL authentication with Azure AD trying to add prompt parameter to the sign in. It does not work as expected. 

The observed request `https://login.microsoftonline.com/common/oauth2/v2.0/authorize?...` doesn't contain the prompt parameter and the request just completes with SSO.

### Reproduction Steps

RedirectToLoginSelectAccount.razor
```
@using Microsoft.AspNetCore.Components.WebAssembly.Authentication;
@inject NavigationManager Navigation

@code {
    protected override void OnInitialized()
    {
        var requestOptions = new InteractiveRequestOptions()
            {
                Interaction = InteractionType.SignIn,
                ReturnUrl = Navigation.Uri,
            };

        requestOptions.TryAddAdditionalParameter("prompt", "select_account");

        Navigation.NavigateToLogin("authentication/login", requestOptions);
    }
}
```

The following weird version works on development build in VS, but doesn't work in published release build.
```
        var extraQueryParametersDictionary = new Dictionary<string, string> { ["prompt"] = "select_account" };
        requestOptions.TryAddAdditionalParameter("extraQueryParameters",
            JsonSerializer.Deserialize<JsonElement>(JsonSerializer.Serialize(extraQueryParametersDictionary)));
```

### Expected behavior

Authentication to include the prompt parameter.

### Actual behavior

Prompt parameter missing.

### Configuration

7.0.101 [C:\Program Files\dotnet\sdk]

    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="7.0.0-rtm.22512.1" />
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="7.0.0-rtm.22512.1" PrivateAssets="all" />
    <PackageReference Include="Microsoft.Authentication.WebAssembly.Msal" Version="7.0.0-rtm.22512.1" />
    <PackageReference Include="Microsoft.Extensions.Http" Version="7.0.0-rtm.22511.4" />

**Comment by juho-hanhimaki** [comment]
I am pretty sure this is a bug. Hopefully someone can take a look and reproduce. I don't think I am able to test using the newer rtm.22513.3 packages as they complain about downgrades from 7.0.0 (which isn't out yet).

https://github.com/dotnet/aspnetcore/issues/42580

https://learn.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/additional-scenarios?view=aspnetcore-7.0#customize-the-login-process

**Comment by TanayParikh** [comment]
We're also using that workaround in the test code: https://github.com/dotnet/aspnetcore/blob/2e2972065f59a186259b7d19357951a4874686c4/src/Components/WebAssembly/WebAssembly.Authentication/test/RemoteAuthenticationServiceTests.cs#L419

This appears to be a [known issue](https://youtu.be/YR3O3ktuYCw?t=946), @javiercn do you have a tracking issue for this, or is the workaround ([also mentioned here](https://github.com/dotnet/aspnetcore/issues/25391#issuecomment-1277920970)) the recommended resolution for now? Note for the workaround, @juho-hanhimaki mentioned:

> The following weird version works on development build in VS, but doesn't work in published release build.

**Comment by TanayParikh** [comment]
@javiercn just to confirm the next steps here, were there any changes made following the [community standup](https://youtu.be/YR3O3ktuYCw?t=946) which may not have flowed through / may not have been comprehensive? Or are we just initially investigating?

**Comment by juho-hanhimaki** [comment]
Fyi, seeing this issue still on published release builds on 7.0.0 release.

**Comment by javiercn** [comment]
@TanayParikh this is worth investigating.

There are some issues I think with the way we do Json serialization internally and some linking related issues.

**Comment by javiercn** [comment]
I believe this is likely caused by the same issue as https://github.com/dotnet/aspnetcore/issues/44973

**Comment by javiercn** [comment]
@juho-hanhimaki try disabling linking for this assembly

Add this to your csproj
```
<ItemGroup>
  <TrimmerRootDescriptor Include="TrimmerRootDescriptor.xml" />
</ItemGroup>
```
And this is TrimmerRootDescriptor.xml:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<linker>
	<assembly fullname="Microsoft.Authentication.WebAssembly.Msal" preserve="all" />
	<assembly fullname="Microsoft.AspNetCore.Components.WebAssembly.Authentication" preserve="all" />
</linker>
```

**Comment by juho-hanhimaki** [comment]
@javiercn that does solve the problem.

**Comment by javiercn** [comment]
@juho-hanhimaki thanks for the confirmation.

This looks very much like the same issue then. Hopefully we can get a fix into the next patch.

**Comment by javiercn** [comment]
this was addressed in 7.0.1 as part of https://github.com/dotnet/aspnetcore/pull/45028
